contracts/ColonyTask.sol

```diff
@@ -310,15 +310,16 @@ contract ColonyTask is ColonyStorage, DSMath {
     Task storage task = tasks[_id];
     IColonyNetwork colonyNetworkContract = IColonyNetwork(colonyNetworkAddress);
 
+    task.finalized = true;
+
     for (uint8 roleId = 0; roleId <= 2; roleId++) {
       uint payout = task.payouts[roleId][token];
       Role storage role = task.roles[roleId];
 
       uint8 rating = (roleId == EVALUATOR) ? 50 : role.rating;
-      int divider = (roleId == WORKER) ? 30 : 50;
 
-      int reputation = SafeMath.mulInt(int(payout), (int(rating)*2 - 50)) / divider;
-      colonyNetworkContract.appendReputationUpdateLog(role.user, reputation, task.domainId);
+      int reputation = SafeMath.mulInt(int(payout), (int(rating)*2 - 50)) / ((roleId == WORKER) ? 30 : 50);
+      colonyNetworkContract.appendReputationUpdateLog(role.user, reputation, domains[task.domainId].skillId);
```

Comment:
Not surprised that this has broken some of the reputation mining tests but shouldn't be too hard to fix